core/src/network/query.rs

```diff
@@ -2,14 +2,26 @@ use holochain_core_types::{crud_status::CrudStatus, entry::EntryWithMetaAndHeade
 use holochain_json_api::{error::JsonError, json::JsonString};
 use holochain_persistence_api::cas::content::Address;
 
+#[derive(Debug, Serialize, Deserialize, PartialEq, DefaultJson, Clone)]
+pub enum GetLinksNetworkQuery {
+    Count,
+    Links,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, DefaultJson, Clone)]
+pub enum GetLinksNetworkResult {
+    Count(usize),
+    Links(Vec<(Address, CrudStatus)>),
+}
+
 #[derive(Debug, Serialize, Deserialize, PartialEq, DefaultJson, Clone)]
 pub enum NetworkQuery {
     GetEntry,
-    GetLinks(String, String),
+    GetLinks(String, String, Option<CrudStatus>, GetLinksNetworkQuery),
```

Comment:
I regard link counts as a different query, instead of a parameter to `GetLinks`. I think it would be much cleaner to represent it here in that way too: instead of introducing those enums above, why don't we have a `GetLinksCount(String, String)` specialization here?